File: js/cat.js
var assert = require('assert');

process.stdin.setEncoding('utf8');
process.stdout.setEncoding('utf8');

var input = '';

process.stdin.on('data', function(data) { input += data; });
process.stdin.on('end', function() { handleInput(); });
process.stdin.resume();

function handleInput() {
  var root = JSON.parse(input);
  assert.equal(root.shift(), 'text');

  root.forEach(function(block) {
    var blockType = block.shift();
    var blockAttr = block.shift();

    block.forEach(function(span) {
      var spanType = span.shift();
      assert(span.length, 1);
      var text = span[0];
      process.stdout.write(text);
    });

    process.stdout.write('\n\n');
  });
};

File: c/cat.c
#include <jansson.h>

int main() {
  json_t *json;
  json_error_t error;
  int i, j;

  json = json_loadf(stdin, 0, &error);

  for (i = 1; i < json_array_size(json); i++) {
    json_t *block = json_array_get(json, i);
    const char *blockType = json_string_value(json_array_get(block, 0));
    json_t *blockAttr = json_array_get(block, 1);

    for (j = 2; j < json_array_size(block); j++) {
      json_t *span = json_array_get(block, j);
      const char *spanType = json_string_value(json_array_get(span, 0));
      const char *text = json_string_value(json_array_get(span, 1));
      printf("%s", text);
    }
    printf("\n\n");
  }

  json_decref(json);

  return 0;
};
